A29-xds-tls-security.md

```diff
@@ -328,93 +350,101 @@ gRPC does not support the other certificate acquisition mechanisms specified by
 * [validation_context_sds_secret_config][VAL-SDS]
 * [validation_context_certificate_provider][VAL-PROVIDER]
 
-When fields of type [`CertificateProviderInstance`][CPI] are present gRPC ignores the respective
+When fields of type [`CertificateProviderPluginInstance`][CPPI] (or the deprecated
```

Comment:
> These next two paragraphs are confusing. They also seem overly broad, because I think this applies only to the client side, not to the server side; on the server side, we can simply NACK if the `tls_certificate_provider_instance` field is unset, regardless of whether any of the unsupported fields are set.

On the server side we have the issue of validation_context and the control plane sending us only the unsupported field(s). Something like this:

```
            "combinedValidationContext": {
              "defaultValidationContext": {
              },
              "validationContextSdsSecretConfig": {
                "name": "validation_context_sds",
                "sdsConfig": {
                  "path": "/etc/envoy/validation_context_sds_secret.yaml"
                }
              }
            }
```

Instead of sending something like this:

```
            "combinedValidationContext": {
              "defaultValidationContext": {
                 "caCertificateProviderInstance": {
                    "instanceName": "google_cloud_private_spiffe",
                    "certificateName": "ROOTCA"
                 } 
              },
              "validationContextSdsSecretConfig": {
                "name": "validation_context_sds",
                "sdsConfig": {
                  "path": "/etc/envoy/validation_context_sds_secret.yaml"
                }
              }
            }
```

The first one should be NACKed but the second one we can process. Isn't that true?

Reply:
Your example is correct, but I think that's already covered by the text I suggested above to describe the server-side behavior:

"""
If a secure server is configured for mTLS, it will need configuration for how to validate the client's certificate.  This validation is configured via the `CertificateValidationContext` message, which is present in either the `validation_context` field or in the `default_validation_context` field inside of the `combined_validation_context` field.  If neither of those fields are set, the server will not request the client's certificate during the TLS handshake (i.e., it will use TLS instead of mTLS).

If a `CertificateValidationContext` is provided, then the server requires a CA root certificate to be able to validate the client certificate.  The CA root certificate is obtained using the certificate provider instance configured via the `ca_certificate_provider_instance` field.  If this field is not present, or if it specifies a certificate provider instance that is not configured in the bootstrap file, gRPC will NACK the LDS update.
"""

More generally, I think the document will be easier to understand if you document the requirements for the client and server separately rather than trying to combine them.